src/views/Home.jsx

```diff
@@ -1,18 +1,52 @@
 import { SingleList } from '../components/SingleList';
 import './Home.css';
+import { createList } from '../api/firebase';
+import { useState } from 'react';
+import { useNavigate } from 'react-router-dom';
+
+export function Home({ data, setListPath, userId, userEmail }) {
+	const [name, setName] = useState('');
+	const navigate = useNavigate();
+
+	const handleSubmit = async (event) => {
+		event.preventDefault();
+		try {
+			const newList = await createList(userId, userEmail, name);
+			setListPath(newList);
+			setName('');
+			alert('The item has been added.');
+			navigate('/list');
+			localStorage.setItem('tcl-shopping-list-path', newList);
```

Comment:
The `setListPath`function saves the token to local storage, so you don't need to do it again with `localStorage.setItem('tcl-shopping-list-path', newList);`

Reply:
okay

Reply:
corrected